diff -urp fluentd-0.12.29.orig/lib/fluent/engine.rb fluentd-0.12.29/lib/fluent/engine.rb
--- fluentd-0.12.29.orig/lib/fluent/engine.rb	2016-09-20 14:50:14.110237370 -0600
+++ fluentd-0.12.29/lib/fluent/engine.rb	2016-09-20 14:51:30.530646968 -0600
@@ -68,8 +68,8 @@ module Fluent
       BasicSocket.do_not_reverse_lookup = true
       Plugin.load_plugins
       if defined?(Encoding)
-        Encoding.default_internal = 'ASCII-8BIT' if Encoding.respond_to?(:default_internal)
-        Encoding.default_external = 'ASCII-8BIT' if Encoding.respond_to?(:default_external)
+        Encoding.default_internal = ENV['DEFAULT_ENCODING'] || 'UTF-8' if Encoding.respond_to?(:default_internal)
+        Encoding.default_external = ENV['DEFAULT_ENCODING'] || 'UTF-8' if Encoding.respond_to?(:default_external)
       end
 
       suppress_interval(system_config.emit_error_log_interval) unless system_config.emit_error_log_interval.nil?
